edfio/edf.py

```diff
@@ -745,7 +745,9 @@ def _set_num_data_records_with_signals(
         if not signals:
             num_data_records = 1
         else:
-            signal_durations = [len(s._digital) / s.sampling_frequency for s in signals]
+            signal_durations = [
+                round(len(s._digital) / s.sampling_frequency, 12) for s in signals
+            ]
```

Comment:
EDF files can have non-integer durations, therefore rounding the duration is not sufficient.

Reply:
Rounding is performed to 12 decimal digits and [test_write_edf_with_non_integer_seconds_duration](https://github.com/the-siesta-group/edfio/blob/cfc43ea7c395774ea1a177d318de13f706b6c0f7/tests/test_edf.py#L688) covers non-integer durations, so that should be okay.